src/kup/nix.py

```diff
@@ -261,7 +270,7 @@ def install_substituter(name: str, substituter: str, pub_key: str) -> None:
 # The `GC_DONT_GC` simply disables the garbage collector used during evaluation of a nix
 # expression. This may cause the process to run out of memory, but hasn't been observed for our
 # derivations in practice, so should be ok to do.
-def nix(args: List[str], is_install: bool = True) -> bytes:
+def nix(args: List[str], is_install: bool = True, exit_on_error: bool = True) -> bytes:
     global CONTAINS_SUBSTITUTERS
```

Comment:
This was added because of these linter errors:

```
src/kup/nix.py:274:27: F823 local variable 'CONTAINS_SUBSTITUTERS' defined in enclosing scope on line 89 referenced before assignment
src/kup/nix.py:276:10: N806 variable 'CONTAINS_SUBSTITUTERS' in function should be lowercase
```

Reply:
Oh, I see. That's because in line 277 you actually do write the variable, so `global` is indeed necessary.